File: frontend/src/services/purchases.ts
import api from './api';

export interface MaterialPurchaseCreate {
  material_id: number;
  quantity: number;
  unit_cost: number;
  supplier_id?: number;
  purchase_date: string; // ISO date string
  notes?: string;
}

export interface MaterialPurchase {
  purchase_id: number;
  company_id: number;
  material_id: number;
  material_name: string;
  quantity: number;
  unit_cost: number;
  total_cost: number;
  supplier_id?: number;
  supplier_name?: string;
  purchase_date: string;
  notes?: string;
  created_at: string;
}

export const purchasesService = {
  async createPurchase(data: MaterialPurchaseCreate): Promise<MaterialPurchase> {
    const response = await api.post('/purchases/', data);
    return response.data;
  },

  async listPurchases(skip = 0, limit = 100): Promise<MaterialPurchase[]> {
    const response = await api.get('/purchases/', {
      params: { skip, limit }
    });
    return response.data;
  },
};

File: backend/app/schemas/purchases.py
from pydantic import BaseModel, Field
from datetime import date
from decimal import Decimal


class MaterialPurchaseCreate(BaseModel):
    """Create a general material purchase (adds to inventory)"""
    material_id: int
    quantity: Decimal = Field(gt=0)
    unit_cost: Decimal = Field(ge=0)
    supplier_id: int | None = None
    purchase_date: date
    notes: str | None = None


class MaterialPurchaseResponse(BaseModel):
    """Material purchase response"""
    id: int = Field(serialization_alias='purchase_id')
    company_id: int
    material_id: int
    material_name: str
    quantity: Decimal
    unit_cost: Decimal
    total_cost: Decimal
    supplier_id: int | None
    supplier_name: str | None
    purchase_date: date
    notes: str | None
    created_at: str
    
    class Config:
        from_attributes = True
